### ISSUE TITLE
Entity Gift Events

### ISSUE THREAD

**Issue by vinderguy** [PaperMC/Paper#8180]
### Is your feature request related to a problem?

Since `LootGenerateEvent` doesn't fire for villager gifts and cat morning gifts, I don't see a way to change the loot (or remove it entirely) other than overriding `gameplay/hero_of_the_village/...` and `gameplay/cat_morning_gift` loot tables via a data pack.

### Describe the solution you'd like.

Adding `VillagerGiftEvent` and `CatMorningGiftEvent` events should solve this issue.

### Describe alternatives you've considered.

Make `LootGenerateEvent` trigger for all kinds of loot generation, including entity deaths, block breaks, and other cases where loot tables are used. Currently `LootGenerateEvent` is called only when generating chest loot.

### Other

If there is a workaround (other than using a data pack) to modify villager and cat gifts, I'd be grateful if you let me know.

**Comment by Doc94** [comment]
I dont check but this things are not handle by https://jd.papermc.io/paper/1.19/org/bukkit/event/entity/EntityDropItemEvent.html ?

**Comment by vinderguy** [comment]
> I dont check but this things are not handle by https://jd.papermc.io/paper/1.19/org/bukkit/event/entity/EntityDropItemEvent.html ?

Thanks! I completely forgot about the [`EntityDropItemEvent`](https://jd.papermc.io/paper/1.19/org/bukkit/event/entity/EntityDropItemEvent.html). I've just checked this, and you're right. It triggers every time a villager or cat gives the player a gift. However, without some complex checks, it is hard to define the context in which items are dropped. For example, the same event also gets called when villagers throw food at each other.

**Comment by Doc94** [comment]
Yeah.. not sure if a feature can be a extended event from [EntityDropItemEvent](https://jd.papermc.io/paper/1.19/org/bukkit/event/entity/EntityDropItemEvent.html) for this cases or give a reason to this event...

**Comment by Machine-Maker** [comment]
I'd rather make a better loot table roll event, with the better loot context API from https://github.com/PaperMC/Paper/pull/7655.
